data_model/src/pagination.rs

```diff
@@ -60,7 +60,7 @@ impl<I: Iterator> Iterator for Paginated<I> {
         #[allow(clippy::option_if_let_else)]
         // Required because of E0524. 2 closures with unique refs to self
         if let Some(start) = self.pagination.start.take() {
-            self.iter.nth(start)
+            self.iter.nth(start as usize)
```

Comment:
I think every such place with `as usize` should be rewritten as `try_from(...).expect("This can't fail because ...")`